Code-the-Dream-School/hh-team4-back
# Dispense medication

## Description

Created A dispense function, and a dispenseLog model. 

## Type of Changes

new feature

Dispense Medication
![Screenshot 2024-12-20 160749](https://github.com/user-attachments/assets/a1240c62-ec9e-442d-bf8e-99ca45d8ed2b)
![dispense medication](https://github.com/user-attachments/assets/91ee22b6-cbd7-469f-bde1-5ad9ddc98235)

get dispense log
![get dispense log stats](https://github.com/user-attachments/assets/5d37a0ae-66d5-4587-a575-40ed718bf320)
![get dispense log controller](https://github.com/user-attachments/assets/14541b04-087c-4de9-8734-21c14e8fac08)

endpoint in post man for dispense medication (note: body)
![Screenshot 2024-12-20 160805](https://github.com/user-attachments/assets/4e3b5b17-3f1d-4cbb-a4b0-f17056de3e9e)

endpoint for fetch dispense log
![Screenshot 2024-12-20 161711](https://github.com/user-attachments/assets/b78b8532-1e14-4d18-8ef9-aab7561336e5)

## Testing Instructions
check postman screen shot above.

Some things to note...
Both controllers for "dispense medication" and for "retrieve dispense log" are located in the medication controller, for the dispense medication controller I have the store comparison commented out, as I was having trouble to getting the users store value to pop up. Also at the bottom of that controller there is the the  DispenseLog.create, which I was not able to test yet. But I can confirm that the rest of the functionality works. aka finding the med, subtracting the med, saving the meds new value. 

I also had to create a DispenseLog model , Which I have created, but also haven't fully tested yet.
